src/main/kotlin/edu/byu/mealplanningassistant/models/Recipe.kt

```diff
@@ -4,17 +4,51 @@ import java.util.*
 
 /**
  * @author Chris West (Chris4942)
- *
- * data makes it so the compiler implements a toString() method and an equals() method for us
- * Putting ()s after the class name allows for defining a default constructor
- * "val"s and "var"s defined in the primary constructor will become member variables on the object
- * "val"s and "var"s are public by default (but "val"s are read only, so there's no issue with that) so there is usually no need for getters and setters
- * It is GOOD practice to avoid using "var"s whenever possible
  */
 data class Recipe (
-    /** Recipe ids should follow the form R_<RandomStringOfChars> */
+    /** Recipe ids should follow the form R_<RandomStringOfChars> and will be randomly generated by the backend */
     val id: String,
-    val name: String, // <val|var> <varName> : <Type>
-    val instructions: String,
-    val attributes: SortedSet<String>, // val means that the variable is final. Use var for variables that need to change
+    val name: String,
+    /** Refers to a username */
+    val owner: String,
+    /** date the recipe was put into the db */
+    val date: Long,
+    val ingredients: Map<String, Amount>,
```

Comment:
Could we just have the amount be a string?

Reply:
(like do we need to have specific unit and quantity) -- sometimes the recipe calls for "3 apples" or "a pinch of garlic salt" so I'm thinking it might get complicated to try to map everything to the amount class but we definitely could do it.